client/src/main/java/org/threadly/litesockets/client/http/HTTPClient.java

```diff
@@ -0,0 +1,406 @@
+ * 
+ */
+public class HTTPClient {
+  public static final int DEFAULT_CONCURRENT = 2;
+  public static final int DEFAULT_TIMEOUT = 15000;
+  public static final ByteBuffer EMPTY_BUFFER = ByteBuffer.allocate(0); 
+  public static final int MAX_HTTP_RESPONSE = 1048576;  //1MB
+  
+  private final AtomicBoolean isRunning = new AtomicBoolean(false);
+  private final int maxResponseSize;
+  private final SubmitterScheduler ssi;
+  private final SocketExecuter sei;
+  private final ConcurrentLinkedQueue<HTTPRequestWrapper> queue = new ConcurrentLinkedQueue<HTTPRequestWrapper>();
+  private final ConcurrentHashMap<TCPClient, HTTPRequestWrapper> inProcess = new ConcurrentHashMap<TCPClient, HTTPRequestWrapper>();
+  private final ConcurrentHashMap<HTTPAddress, LinkedList<TCPClient>> sockets = new ConcurrentHashMap<HTTPAddress, LinkedList<TCPClient>>();
+  private final MainClientProcessor mcp = new MainClientProcessor();
+  private final int maxConcurrent;
+  private volatile int defaultTimeout = DEFAULT_TIMEOUT;
+
+  private NoThreadSocketExecuter ntse = null;
+  private SingleThreadScheduler sts = null;
+
+  /**
+   * <p>This is the default constructor it will create its own {@link SingleThreadScheduler} to use as a threadpool, as
+   * well as using the default {@value DEFAULT_CONCURRENT} and {@value MAX_HTTP_RESPONSE}.  This means we will do at most
+   * 2 HTTPRequests at the same time, and those responses can be up to 1mb in size.</p>
+   * 
+   */
+  public HTTPClient() {
+    this(DEFAULT_CONCURRENT, MAX_HTTP_RESPONSE);
+  }
+  
+  /**
+   * <p>This constructor will let you set the max Concurrent Requests and max Response Size but will still 
+   * create its own {@link SingleThreadScheduler} to use as a threadpool.</p> 
+   * 
+   */
+  public HTTPClient(int maxConcurrent, int maxResponseSize) {
+    this.maxConcurrent = maxConcurrent;
+    this.maxResponseSize = maxResponseSize;
+    sts = new SingleThreadScheduler();
+    this.ssi = sts;
+    ntse = new NoThreadSocketExecuter();
+    sei = ntse;
+    ntse.start();
+  }
+  
+  
+  /**
+   * <p>This constructor will let you set the max Concurrent Requests and max Response Size
+   * as well as your own {@link SocketExecuter} as the thread pool to use.</p> 
+   * 
+   */
+  public HTTPClient(int maxConcurrent, int maxResponseSize, SocketExecuter sei) {
+    this.maxConcurrent = maxConcurrent;
+    this.maxResponseSize = maxResponseSize;
+    this.ssi = sei.getThreadScheduler();
+    this.sei = sei;
+  }
+  
+  public HTTPResponseData request(URL url) throws HTTPParsingException {
+    HTTPResponseData hr = null;
+    try {
+      hr = requestAsync(url).get();
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+    } catch (Exception e) {
+      if(e.getCause() instanceof HTTPParsingException) {
+        throw (HTTPParsingException)e.getCause();
+      } else {
+        throw new HTTPParsingException(e.getCause());
```

Comment:
Seems like in this case you want to just put e as the cause, unless your catching specifically an "ExecutionException", things like a cancellation would be lost otherwise